weave/tests/test_weave_client.py

```diff
@@ -1165,3 +1166,96 @@ def test_table_partitioning(network_proxy_client):
         + 1  # the second  create
         + NUM_ROWS / 2  # updates - 2 per batch
     )
+
+
+def test_summary_tokens_cost(client):
+    is_sqlite = isinstance(client.server._internal_trace_server, SqliteTraceServer)
+    if is_sqlite:
+        # SQLite does not support costs
+        return
+
+    @weave.op()
+    def model_a(text):
+        result = "a: " + text
+        return {
+            "result": result,
+            "model": "model_a",
+            "usage": {
+                "prompt_tokens": len(text),
+                "completion_tokens": len(result),
+            },
+        }
+
+    @weave.op()
+    def model_b(text):
+        result = "bbbb: " + text
+        return {
+            "result": result,
+            "model": "model_b",
+            "usage": {
+                "prompt_tokens": len(text),
+                "completion_tokens": len(result),
+            },
+        }
+
+    @weave.op()
+    def models(text):
+        return (
+            model_a(text)["result"]
+            + " "
+            + model_a(text)["result"]
+            + " "
+            + model_b(text)["result"]
+        )
+
+    res = models("hello")
+    assert res == "a: hello a: hello bbbb: hello"
+
+    call = list(models.calls())[0]
+
+    assert call.summary["usage"] == {
+        "model_a": {"requests": 2, "prompt_tokens": 10, "completion_tokens": 16},
+        "model_b": {"requests": 1, "prompt_tokens": 5, "completion_tokens": 11},
+    }
+
+    callsWithCost = list(client.calls(include_costs=True))
+    callsNoCost = list(client.calls(include_costs=False))
+
+    assert len(callsWithCost) == len(callsNoCost)
+    assert len(callsWithCost) == 4
+
+    noCostCallSummary = callsNoCost[0].summary
+    withCostCallSummary = callsWithCost[0].summary
+
+    assert withCostCallSummary.get("weave", "bah") != "bah"
+    assert len(withCostCallSummary["weave"]["costs"]) > 0
```

Comment:
is this the only place where you are validating that your cost calculation is correct? Are there any other places where maybe you test:
* mixed models
* models that have 2 entries in the cost table over time,
* ...

The primary end-results?